runtime/v2/shim/publisher.go

```diff
@@ -128,9 +128,12 @@ func (l *RemoteEventsPublisher) Publish(ctx context.Context, topic string, event
 }
 
 func (l *RemoteEventsPublisher) forwardRequest(ctx context.Context, req *v1.ForwardRequest) error {
+	fCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
 	service, err := l.client.EventsService()
 	if err == nil {
-		_, err = service.Forward(ctx, req)
+		_, err = service.Forward(fCtx, req)
```

Comment:
The `ctx` isn't used in between here and next Forward, this could avoid the defer by just calling cancel after.

Reply:
Fixed.